client/analytics/report/revenue.js

```diff
@@ -212,13 +220,68 @@ class RevenueReport extends Component {
 		];
 	}
 
-	render() {
-		const { path, query } = this.props;
-		const { totals = {}, intervals = [] } = this.state.stats;
+	renderTable() {
+		const {
+			tableData,
+			isTableDataRequesting,
+			isTableDataError,
+			query,
+			previousTableData,
+		} = this.props;
+		const headers = this.getHeadersContent();
+
+		if ( isTableDataRequesting && isReportDataEmpty( previousTableData ) ) {
+			return (
+				<TableCardPlaceholder
+					rows={ 10 }
+					headers={ headers }
+					title={ __( 'Revenue Last Week', 'wc-admin' ) }
+				/>
+			);
+		}
+
+		// TODO Update Error Display
+		if ( isTableDataError ) {
+			return (
+				<div className="woocommerce-analytics__report-table-error">
+					{ __( 'There was an error retrieving your revenue summary. Please try again.' ) }
+				</div>
+			);
+		}
+
+		if ( isReportDataEmpty( tableData ) && ! isTableDataRequesting ) {
+			// TODO Return empty content component once it is built.
+			return null;
+		}
+
+		const data = isTableDataRequesting ? previousTableData : tableData;
+		const { totals = {}, intervals = [] } = ( data && data.data ) || {};
+		const { totalResults } = data;
 		const summary = this.getSummaryContent( totals ) || [];
 		const rows = this.getRowsContent( intervals ) || [];
-		const headers = this.getHeadersContent();
 
+		const className = classnames( 'woocommerce-analytics__report-table-card', {
+			'is-requesting': isTableDataRequesting,
+		} );
+
+		return (
+			<TableCard
+				title={ __( 'Revenue Last Week', 'wc-admin' ) }
```

Comment:
This seems to be fetching all of my report data, not just within the last week. The pagination & column sorting approach we've taken assume an "all revenue data (per day)" functionality, so I think we just need to relabel this table. Thoughts @woocommerce/woo-core-design  ?

Reply:
Yeah good catch. This table is not just for the last week - it lets you page through all results. Should just this just be "Revenue"?